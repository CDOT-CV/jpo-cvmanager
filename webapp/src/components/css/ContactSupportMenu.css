#ContactSupportMenu {
    position: absolute;
    bottom: 5%;
    right: 5%;

    @media only screen and (max-height: 800px) {
        transform: translate(150%, 150%);
    }
}

h5 {
    font-size: 1.5rem;
    font-weight: 400;
    margin-bottom: 0.5rem;
    text-align: center;
}

.label {
    font-size: 1rem;
    font-weight: 400;
    margin-bottom: 0.5rem;
}

.btn {
    font-size: 1rem;
    font-weight: 400;
    margin-bottom: 0.5rem;
}

.mb-3 {
    margin-bottom: 1rem !important;
}