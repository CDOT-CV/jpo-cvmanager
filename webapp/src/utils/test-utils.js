/**
 * This function searches for the every react-aria SSR ids in a given HTMLElement node and replace every attribute values with a static id
 *
 * This can be usefull when you're trying to generate a snapshot of components using react-aria under the hood
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceReactAriaIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: "class",
      updateFunc: (val) =>
        val.replace(
          /css-[0-9a-z]{6}-MuiTableCell-root-MuiTablePagination-root/g,
          "css-mocked-MuiTableCell-root-MuiTablePagination-root"
        ),
    },
  ];

  container.querySelectorAll("td").forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)));
      }
    });
  });
  container.querySelectorAll("input[aria-invalid]").forEach((item) => {
    delete item["aria-invalid"];
  });
  return container;
}

export { replaceChaoticIds };
