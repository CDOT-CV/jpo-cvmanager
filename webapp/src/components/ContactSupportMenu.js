import React, { useState } from "react";
import { Form } from "react-bootstrap";
import { useForm } from "react-hook-form";

import "../components/css/Admin.css";
import "react-widgets/styles.css";
import EnvironmentVars from "../EnvironmentVars";

const ContactSupportMenu = () => {
    const [successMsg, setSuccessMsg] = useState("");
    const [errorState, setErrorState] = useState(false);
    const [errorMessage, setErrorMessage] = useState("");
    const {
        register,
        handleSubmit,
        reset,
        formState: { errors },
    } = useForm();

    const sendPostData = async (json) => {
        
        try {
            const res = await fetch(EnvironmentVars.sendEmail, {
                method: "POST",
                headers: {
                    "Content-Type": "application/json",
                    "Authorization": "Bearer " + localStorage.getItem("token"),
                },
                body: JSON.stringify(json),
            });

            const status = res.status;
            const data = await res.json();
            if (status === 200) {
                console.debug("Successfully sent email");
                setSuccessMsg("Successfully sent email");
                setErrorState(false);
                reset();
            }
            else if (status === 500) {
                console.error(data);
                setSuccessMsg("");
                setErrorState(true);
                setErrorMessage(data);
            }
        } catch (exception_var) {
            console.error(exception_var);
            setSuccessMsg("");
            setErrorState(true);
            setErrorMessage(exception_var);
        }      
    };

    const onSubmit = (data) => {
        sendPostData(data);
    };

    return (
        <div>
            <Form onSubmit={handleSubmit(onSubmit)}>
                <h5>Contact Support</h5>
                <Form.Group className="mb-3" controlId="email">
                    <Form.Label>Your Email</Form.Label>
                    <Form.Control
                        type="email"
                        placeholder="Enter your email"
                        {...register("email", {
                            required: "Email is required",
                        })}
                    />
                    {errors.email && (
                        <Form.Text className="text-danger">
                            {errors.email.message}
                        </Form.Text>
                    )}
                </Form.Group>
                <Form.Group className="mb-3" controlId="subject">
                    <Form.Label>Subject</Form.Label>
                    <Form.Control
                        type="text"
                        placeholder="Enter your subject"
                        {...register("subject", {
                            required: "Subject is required",
                        })}
                    />
                    {errors.subject && (
                        <Form.Text className="text-danger">
                            {errors.subject.message}
                        </Form.Text>
                    )}
                </Form.Group>
                <Form.Group className="mb-3" controlId="message">
                    <Form.Label>Message</Form.Label>
                    <Form.Control
                        as="textarea"
                        rows={5}
                        placeholder="Enter your message"
                        {...register("message", {
                            required: "Message is required",
                        })}
                    />
                    {errors.message && (
                        <Form.Text className="text-danger">
                            {errors.message.message}
                        </Form.Text>
                    )}
                </Form.Group>

                {successMsg && <p className="success-msg">{successMsg}</p>}
                {errorState && (
                    <p className="error-msg">
                        Failed to send email due to error: {errorMessage}
                    </p>
                )}
                <div className="form-control">
                    <label></label>
                    <button type="submit" className="btn btn-primary">
                        Send Email
                    </button>
                </div>
            </Form>
        </div>
    );
}

export default ContactSupportMenu;