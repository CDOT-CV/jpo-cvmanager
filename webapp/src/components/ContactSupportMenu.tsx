import React, { useState } from 'react'
import { Form } from 'react-bootstrap'
import { useForm } from 'react-hook-form'

import 'react-widgets/styles.css'
import RsuApi from '../apis/rsu-api'

import './css/ContactSupportMenu.css'
import toast from 'react-hot-toast'

const ContactSupportMenu = () => {
  const [hidden, setHidden] = useState(true) // hidden by default
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm()

  const onSubmit = async (data: Object) => {
    try {
      const res = await RsuApi.postContactSupport(data)
      const status = res.status
      if (status === 200) {
        toast.success('Successfully sent email')
        reset()
      } else {
        toast.error('Something went wrong: ' + status)
      }
    } catch (exception_var) {
      toast.error('An exception occurred, please try again later')
    }
  }

  if (hidden) {
    return (
      <div>
        <button
          type="button"
          className="showbutton"
          onClick={() => {
            setHidden(!hidden)
          }}
        >
          Contact Support
        </button>
      </div>
    )
  }

  return (
    <div id="ContactSupportMenu">
      <Form onSubmit={handleSubmit(onSubmit)}>
        <h5>Contact Support</h5>
        <Form.Group className="mb-3" controlId="email">
          <Form.Label className="label">Your Email</Form.Label>
          <Form.Control
            type="email"
            placeholder="Enter your email (Required)"
            {...register('email', {
              required: 'Email is required',
            })}
          />
          {errors.email && <Form.Text className="text-danger">{errors.email.message}</Form.Text>}
        </Form.Group>
        <Form.Group className="mb-3" controlId="subject">
          <Form.Label className="label">Subject</Form.Label>
          <Form.Control
            type="text"
            placeholder="Enter your subject (Required)"
            {...register('subject', {
              required: 'Subject is required',
            })}
          />
          {errors.subject && <Form.Text className="text-danger">{errors.subject.message}</Form.Text>}
        </Form.Group>
        <Form.Group className="mb-3" controlId="message">
          <Form.Label className="label">Message</Form.Label>
          <Form.Control
            as="textarea"
            rows={5}
            placeholder="Enter your message (Required)"
            {...register('message', {
              required: 'Message is required',
            })}
          />
          {errors.message && <Form.Text className="text-danger">{errors.message.message}</Form.Text>}
        </Form.Group>
        <div className="form-control">
          <label></label>
          <button type="submit" className="btn btn-primary">
            Send Email
          </button>
        </div>
        <div>
          <button
            type="button"
            className="hidebutton"
            onClick={() => {
              setHidden(!hidden)
            }}
          >
            x
          </button>
        </div>
      </Form>
    </div>
  )
}

export default ContactSupportMenu
