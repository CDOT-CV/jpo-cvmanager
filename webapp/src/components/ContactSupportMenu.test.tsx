import React from 'react'
import { render } from '@testing-library/react'
import ContactSupportMenu from './ContactSupportMenu'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
    const { container } = render(<ContactSupportMenu />)
    
    expect(replaceChaoticIds(container)).toMatchSnapshot()
})