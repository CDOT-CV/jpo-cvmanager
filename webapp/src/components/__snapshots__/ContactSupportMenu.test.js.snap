// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div>
    <form
      class=""
    >
      <h5>
        Contact Support
      </h5>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="email"
        >
          Your Email
        </label>
        <input
          class="form-control"
          id="email"
          name="email"
          placeholder="Enter your email"
          type="email"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="subject"
        >
          Subject
        </label>
        <input
          class="form-control"
          id="subject"
          name="subject"
          placeholder="Enter your subject"
          type="text"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="message"
        >
          Message
        </label>
        <textarea
          class="form-control"
          id="message"
          name="message"
          placeholder="Enter your message"
          rows="5"
        />
      </div>
      
      <div
        class="form-control"
      >
        <label />
        <button
          class="btn btn-primary"
          type="submit"
        >
          Send Email
        </button>
      </div>
    </form>
  </div>
</div>
`;
